collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -566,7 +573,7 @@ private void setupViewModels(FormEntryViewModelFactory formEntryViewModelFactory
     }
 
     private void formControllerAvailable(@NonNull FormController formController) {
-        formSessionRepository.set(sessionId, formController);
+        formSessionRepository.set(sessionId, formController, formsRepository.getOneByPath(formPath));
```

Comment:
I was wondering if we could avoid the extra DB lookup here, but it seems like that would be awkward. I'd hope that eventually the whole form load (loading from the DB and creating the JavaRosa objects) will happen in one neat place that could then set the form session after it's done.

Reply:
Yeah, I was thinking about passing `form`/`instance` objects from `FormUriActivity` to `FormFillingActivity` so that `FormFillingActivity` does not have to deal with that since `FormUriActivity` needs to read the database and validate the output either way. But that's something for the future.